core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -52,9 +61,73 @@ public SigningStage(HttpClientDependencies dependencies) {
     @Override
     public SdkHttpFullRequest execute(SdkHttpFullRequest request, RequestExecutionContext context) throws Exception {
         InterruptMonitor.checkInterrupted();
+        if (context.executionAttributes().getAttribute(SdkExecutionAttribute.SELECTED_AUTH_SCHEME) != null) {
+            return sraSignRequest(request, context);
+        }
         return signRequest(request, context);
     }
 
+    private <T extends Identity> SdkHttpFullRequest sraSignRequest(SdkHttpFullRequest request, RequestExecutionContext context) {
+
+        ExecutionAttributes executionAttributes = context.executionAttributes();
+        SelectedAuthScheme<T> selectedAuthScheme =
+            executionAttributes.getAttribute(SdkExecutionAttribute.SELECTED_AUTH_SCHEME);
+
+        AuthSchemeOption authSchemeOption = selectedAuthScheme.authSchemeOption();
+
+        // TODO: Identity resolution should move to before Endpoint resolution interceptor, to support accountId based endpoints
+        //  and also to logically separate out identity resolution as its own step.
+        ResolveIdentityRequest.Builder identityRequestBuilder = ResolveIdentityRequest.builder();
+        authSchemeOption.forEachIdentityProperty(identityRequestBuilder::putProperty);
+
+        IdentityProvider<T> identityProvider = selectedAuthScheme.identityProvider();
+        T identity = identityProvider.resolveIdentity(identityRequestBuilder.build()).join();
+
+        // TODO: resolveIdentity(Consumer builder) doesn't get used
+
+        // Consumer<ResolveIdentityRequest.Builder> c = r -> r.putProperty(IdentityProperty.create(String.class, "key"), "value");
+        // identityProvider.resolveIdentity(c);
+        // TODO: Consumer Builder is not working because of ambiguity!!
+        // identityProvider.resolveIdentity(r -> r.putProperty(IdentityProperty.create(String.class, "key"), "value"));
+
+        SyncSignRequest.Builder<T> signRequestBuilder = SyncSignRequest
+            .builder(identity)
+            .request(request)
+            .payload(request.contentStreamProvider().orElse(null));
+        authSchemeOption.forEachSignerProperty(signRequestBuilder::putProperty);
+
+        HttpSigner<T> signer = selectedAuthScheme.signer();
+        SyncSignedRequest signedRequest = signer.sign(signRequestBuilder.build());
+
+        // TODO: sign(Consumer builder) doesn't get used
+        // SyncSignedRequest signedRequest =
+        //     signer.sign(r -> r.identity(identity)
+        //                       .request(request)
+        //                       .payload(request.contentStreamProvider().orElse(null))
+        //                       .putProperty(SignerProperty.create(String.class, "key"), "value"));
+
+        return toSdkHttpFullRequest(signedRequest);
+    }
+
+    private SdkHttpFullRequest toSdkHttpFullRequest(SyncSignedRequest signedRequest) {
```

Comment:
I don't know this code well but it seems odd that we must do this conversion - guess because of the new signer APIs. Is it the right place to put it?

Reply:
Conversion needed because output of new signer API needs to be converted to output of SigningStage (SdkHttpFullRequest). I don't want `toSdkHttpFullRequest` in the new http-auth-spi module. A builder/create method inside SdkHttpFullRequest is an option, but that's a SdkPublicApi, so hidden inside this SigningStage is probably better. When the request pipeline is rewritten, it _may_ not need the conversion?

Reply:
+1 when the request pipeline is rewritten I would love to do away with the SdkHttpFullRequest.